### PR TITLE
Added sample code for GA, Localytics, and Amplitude integration.

### PR DESCRIPTION
**Pull request by alda-optimizely**
@elliotykim @wangjoshuah 
I wrote sample code for integrations in Swift. They're all in the demo app. Elliot -- you can use this for the integration docs as well.

### CODE REVIEW

File: OptimizelyDemoApp/AppDelegate.swift [hunk 1]
```diff
@@ -43,6 +46,75 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                let propertyKey : String! = "[Optimizely] " + experiment.experimentKey
+                let identify : AMPIdentify = AMPIdentify()
+                identify.set(propertyKey, value:variation.variationKey as NSObject!)
+                
+                // Track impression event (optional)
+                let eventIdentifier : String = "[Optimizely] " + experiment.experimentKey + " " + variation.variationKey
+                Amplitude.instance().logEvent(eventIdentifier)
+                
+                // ---- Google Analytics ----
+                let tracker : GAITracker? = GAI.sharedInstance().defaultTracker
```
Comment (elliotykim) [inline OptimizelyDemoApp/AppDelegate.swift]: @alda-optimizely are there any imports required for GAI?

File: OptimizelyDemoApp/AppDelegate.swift [hunk 2]
```diff
@@ -43,6 +46,75 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                
+                let action : String = "Experiment - " + experiment.experimentKey
+                let label : String = "Variation - " + variation.variationKey
+                
+                // Build and send a non-interaction Event
+                let builder = GAIDictionaryBuilder.createEvent(withCategory:"Optimizely", action: action, label: label, value: nil).build()
```
Comment (elliotykim) [inline OptimizelyDemoApp/AppDelegate.swift]: add space:
`withCategory: "Optimizely"`

File: OptimizelyDemoApp/AppDelegate.swift [hunk 3]
```diff
@@ -43,6 +46,75 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        if let experiment = userInfo?["experiment"] as! OPTLYExperiment? {
+            if let variation = userInfo?["variation"] as! OPTLYVariation? {
+                
+                // ---- Amplitude ----
+                let propertyKey : String! = "[Optimizely] " + experiment.experimentKey
+                let identify : AMPIdentify = AMPIdentify()
+                identify.set(propertyKey, value:variation.variationKey as NSObject!)
+                
+                // Track impression event (optional)
+                let eventIdentifier : String = "[Optimizely] " + experiment.experimentKey + " " + variation.variationKey
```
Comment (elliotykim) [inline OptimizelyDemoApp/AppDelegate.swift]: add dash:
`"[Optimizely] " + experiment.experimentKey + " - " + variation.variationKey`

Comment (elliotykim) [reply OptimizelyDemoApp/AppDelegate.swift]: To match the other languages

File: OptimizelyDemoApp/AppDelegate.swift [hunk 4]
```diff
@@ -43,6 +46,75 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                
+                let action : String = "Experiment - " + experiment.experimentKey
+                let label : String = "Variation - " + variation.variationKey
+                
+                // Build and send a non-interaction Event
+                let builder = GAIDictionaryBuilder.createEvent(withCategory:"Optimizely", action: action, label: label, value: nil).build()
+                tracker?.send(builder as [NSObject : AnyObject]!)
+                
+                // ---- Mixpanel ----
+                Mixpanel.initialize(token:"MIXPANEL_TOKEN")
```
Comment (elliotykim) [inline OptimizelyDemoApp/AppDelegate.swift]: Probably don't need to initialize Mixpanel every time this listener is calld

File: OptimizelyDemoApp/AppDelegate.swift [hunk 5]
```diff
@@ -43,6 +46,75 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        let attributes : NSMutableDictionary = [:]
+        attributes.addEntries(from: userAttributes!)
+        
+        if let userExperimentVariationMapping = userInfo?["ExperimentVariationMapping"] as! Dictionary<String, AnyObject>? {
+            for (key,value) in userExperimentVariationMapping {
+                let variation : OPTLYVariation = value as! OPTLYVariation
+                attributes.setValue(key, forKey:variation.variationKey)
+            }
+        }
+        // Track impression event (optional)
```
Comment (elliotykim) [inline OptimizelyDemoApp/AppDelegate.swift]: This comment should be "// Tag custom event with attributes"

**Comment by elliotykim** [comment]
@alda-optimizely I didn't know I'd keep adding more and more comments. Next time I'll create a review so the notes are organized

File: OptimizelyDemoApp/AppDelegate.swift (hunk 1 above)
Comment (alda-optimizely) [reply OptimizelyDemoApp/AppDelegate.swift]: no they're integration is weird -- i had to create a bridging header.
i actually was disappointed how hard it was to get GA in the app.
they also don't support tvOS.

**Review by wangjoshuah — APPROVED** [review]

**Comment by elliotykim** [comment]
lgtm thanks!

**Pull request merged**